### PR TITLE
Issue 10254: Improve coach tabs accessibility of Reports Lesson Tab

### LINKED ISSUE #10254: Improve Coach tabs accessibility - "Report-Learners" tabs on a Reports lesson page
In #10212, we embarked on the journey toward improved accessibility of the coach plugin. Part of it is refactoring tabs logic to use new KDS tabs components that are implemented in an accessible manner. This issue is a continuation of that work.

The goal is to use the KDS tabs components for _Report-Learners_ tabs on a lesson page in the coach reports ( _Coach → Reports → Lessons_ → select a lesson → lesson page)

report-learners

## Acceptance criteria

- [ ] The current implementation is refactored to use KDS tabs components (`KTabs` or the combination of `KTabsList` + `KTabsPanel`)
- [ ] The new implementation satisfies [these](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/) and [these](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-manual/) W3C guidelines (this is already provided by KDS tabs components, however in some cases, a bit of additional Kolibri logic may be needed as well - see comments below)
- [ ] The focus ring shows when navigating with a keyboard and stays on the selected tab after pressing enter

## Comments

- KDS tabs components documentation
  - [Tabs page](https://deploy-preview-426--kolibri-design-system.netlify.app/tabs/)
  - [KTabs component](https://deploy-preview-426--kolibri-design-system.netlify.app/ktabs/)
  - [KTabsList component](https://deploy-preview-426--kolibri-design-system.netlify.app/ktabslist/)
  - [KTabsPanel component](https://deploy-preview-426--kolibri-design-system.netlify.app/ktabspanel)
- You can play around with the _Lessons-Quizzes-Groups-Learners_ tabs that are already implemented with KDS components  to experience the expected behavior
example
- In some cases, due to the routing architecture, some additional Kolibri logic may be needed to ensure seamless focus behavior. If you run into issues with the focus ring disappearing after pressing enter on a tab or similar, [useCoachTabs composable](https://github.com/learningequality/kolibri/blob/5806f4ad50db3d96fdd0030d0659861b52a63e67/kolibri/plugins/coach/assets/src/composables/useCoachTabs.js) may help

### PR DESCRIPTION
**Pull request by muditchoudhary**
## Summary

Use KTabs to improve the accessibility of the Reports Lesson Tabs.

Peek 2023-12-09 17-21

## References

Closes #10254

## Reviewer guidance

To review, Go to Coach -->Select a Class --> Reports --> Lesson --> Select a lesson. Then test the tabs with keyboard tabs key.

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by muditchoudhary** [comment]
Hi, @MisRob Another one has been fixed! Please review it.

I think all the mentioned follow-up issues are fixed. Yayy 🎉

File: kolibri/plugins/coach/assets/src/views/reports/ReportsLessonBase.vue [hunk 1]
```diff
@@ -227,6 +271,12 @@
         }
       },
     },
+    $trs: {
+      coachReportsLesson: {
```
Comment (MisRob) [inline kolibri/plugins/coach/assets/src/views/reports/ReportsLessonBase.vue]: Apologies I forgot to mention this in previous work (and probably also use myself in the PR I recommended as a template). We're migrating away from placing strings into components. Could you please move this string to `commonCoachStrings` and import it from there?

In regards to importing the string, we'd appreciate if you could use this fairly new way of using `setup()` https://kolibri-dev.readthedocs.io/en/develop/i18n.html?highlight=translate#common-string-modules. Even though you may see translation mixins being used all over the codebase, whenever possible we'd like to use this pattern for all new code additions.

**Review by MisRob — APPROVED** [review]
> I think all the mentioned follow-up issues are fixed. Yayy 🎉

Yes, amazing!

Thank you @muditchoudhary, code is looking good and it seems to work well too. I added one non-blocking, nice-to-have, suggestion. And I'll also ask @radinamatic to test before we merge.

**Comment by muditchoudhary** [comment]
Okay I'll have a commit soon. Please wait till then before merging.

**Comment by MisRob** [comment]
Thank you @muditchoudhary! I also wanted to mention I won't be able to follow-up here this year as this is my last day before few weeks of time off. Someone else may jump in, but the whole team's availability will too soon be limited until the second week of January. Thank you and hopefully see you next year :)!

**Comment by radinamatic** [comment]
Any reason why is this PR targeted to ``develop`` instead of 0.16, as the previous one that added the same change (as far as I understand) to quiz reports?

**Comment by MisRob** [comment]
> Any reason why is this PR targeted to develop instead of 0.16, as the previous one that added the same change (as far as I understand) to quiz reports?

@radinamatic Yes, the original intention was to target `develop` with all of new work. Majority of the PRs from this group target it. But the previous PR you mentioned was easier to work on the release branch because the place was broken due to some ongoing coach features.

**Comment by MisRob** [comment]
>  the place was broken due to some ongoing coach features

More accurately, a contributor couldn't create data because of broken quiz creation and therefore it was impossible to navigate to the place they needed to update because it assumed existence of at least one quiz, I believe

@muditchoudhary this is not related to your work

**Comment by muditchoudhary** [comment]
There was a conflict. I resolved it

**Comment by muditchoudhary** [comment]
Please review it. I do not exactly understand how we use string from a common module with setup.

**Comment by MisRob** [comment]
Thank you, @muditchoudhary. I've just returned from my vacation and will have a look at it again in the next couple of days.

**Comment by muditchoudhary** [comment]
> Thank you, @muditchoudhary. I've just returned from my vacation and will have a look at it again in the next couple of days.

Okay. Alright. I hope you had a great vacation. Until then I'll look into some new issues. Thank you!

**Comment by MisRob** [comment]
The latest updates are looking good to me, thank you @muditchoudhary. @radinamatic could you please try this out?

**Comment by MisRob** [comment]
Hi @muditchoudhary, we'll get to test and merge this, we just need to prioritize QA of the release branch rather than the `develop` right now. You can consider it accepted :)

**Comment by muditchoudhary** [comment]
> Hi @muditchoudhary, we'll get to test and merge this, we just need to prioritize QA of the release branch rather than the `develop` right now. You can consider it accepted :)

Okay, got it. Thank you for letting me know.

**Comment by radinamatic** [comment]
Installed the EXE asset from this PR on Windows 10 in both Chrome and Firefox, but I'm unable to open a report for a lesson, using neither the keyboard, nor the mouse. Report for quiz apparently opens correctly.

Firefox | Chrome
-- | --
https://github.com/learningequality/kolibri/assets/1457929/514368f3-f51b-4641-be85-df9c92bcaf66 |

**Comment by muditchoudhary** [comment]
> Installed the EXE asset from this PR on Windows 10 in both Chrome and Firefox, but I'm unable to open a report for a lesson, using neither the keyboard, nor the mouse. Report for quiz apparently opens correctly.
> 
> Firefox	Chrome
> https://github.com/learningequality/kolibri/assets/1457929/514368f3-f51b-4641-be85-df9c92bcaf66	

Currently working on this: #10491 I'll check it after completing it.

**Comment by MisRob** [comment]
I think it will have something to do with the new way of using strings I recommended. I see you followed the documentation so if you don't mind, I would have a look @muditchoudhary rather sooner what's going on so we can eventually fix it (we'd like to use strings in this way in other work)

**Comment by muditchoudhary** [comment]
> I think it will have something to do with the new way of using strings I recommended. I see you followed the documentation so if you don't mind, I would have a look @muditchoudhary rather sooner what's going on so we can eventually fix it (we'd like to use strings in this way in other work)

okay 👍

**Comment by MisRob** [comment]
@muditchoudhary We have a typo in documentation. I fixed it in this PR and will open a PR to fix documentation too. @radinamatic you can now test the page, thank you.

**Comment by radinamatic** [comment]
Tested with NVDA in Firefox on Windows 10, looking much better now! 👏🏽 

Keyboard navigation | NVDA speech viewer
-- | --
reports-lessons |

**Review by radinamatic — APPROVED** [review]
Manual QA & A11y testing passed, yey for more accessible tabs in Kolibri! 💯 👏🏽 :shipit: 

Thank you for your contribution @muditchoudhary!

**Comment by MisRob** [comment]
Yey! Thanks @radinamatic and @muditchoudhary!

**Comment by MisRob** [comment]
Docs failure is not related to this PR and should be fixed on develop, so merging.

**Pull request merged**